vm/src/stdlib/time.rs

```diff
@@ -173,6 +197,169 @@ fn time_strptime(
+                if GetThreadTimes(thread, &mut creation_time, &mut exit_time, &mut kernel_time, &mut user_time) == 0 {
+                    return Err(vm.new_os_error("Failed to get clock time".to_owned()));
+                }
+
+                let mut large: ULARGE_INTEGER = std::mem::zeroed();
+                large.u_mut().LowPart = kernel_time.dwLowDateTime;
+                large.u_mut().HighPart = kernel_time.dwHighDateTime;
+                let ktime = *large.QuadPart();
+
+                large.u_mut().LowPart = user_time.dwLowDateTime;
+                large.u_mut().HighPart = kernel_time.dwHighDateTime;
+                let utime = *large.QuadPart();
+
+                Ok(Duration::from_nanos((ktime + utime) * 100))
+            }
+        } else if #[cfg(any(
+            target_os = "android",
+            target_os = "dragonfly",
+            target_os = "freebsd",
+            target_os = "linux",
+            target_os = "openbsd",
+        ))] {
+            let mut time = libc::timespec {
+                tv_sec: 0,
+                tv_nsec: 0,
+            };
+            if unsafe { libc::clock_gettime(libc::CLOCK_THREAD_CPUTIME_ID, &mut time) } == -1 {
+                return Err(vm.new_os_error("Failed to get clock time".to_owned()));
+            }
+            Ok(Duration::new(time.tv_sec as u64, time.tv_nsec as u32))
+        } else if #[cfg(solaris)] {
+            Ok(Duration::from_nanos(unsafe { libc::gethrvtime() }))
+        } else {
+            Err(vm.new_not_implemented_error("thread time unsupported in this system".to_owned()))
+        }
+    }
+}
+
+fn time_thread_time(vm: &VirtualMachine) -> f64 {
+    match get_thread_time(vm) {
+        Ok(v) => v.as_secs_f64(),
+        Err(err) => panic!("Time error: {:?}", err),
+    }
+}
+
+fn time_thread_time_ns(vm: &VirtualMachine) -> u64 {
+    match get_thread_time(vm) {
+        Ok(v) => v.as_nanos() as u64,
+        Err(err) => panic!("Time error: {:?}", err),
+    }
+}
+
+#[cfg(windows)]
+fn get_process_time(vm: &VirtualMachine) -> PyResult<Duration> {
+    let mut creation_time = FILETIME::default();
+    let mut exit_time = FILETIME::default();
+    let mut kernel_time = FILETIME::default();
+    let mut user_time = FILETIME::default();
+    unsafe {
+        let process = GetCurrentProcess();
+        if GetProcessTimes(
+            process,
+            &mut creation_time,
+            &mut exit_time,
+            &mut kernel_time,
+            &mut user_time,
+        ) == 0
+        {
+            return Err(vm.new_os_error("Failed to get clock time".to_owned()));
+        }
+
+        let mut large: ULARGE_INTEGER = std::mem::zeroed();
+        large.u_mut().LowPart = kernel_time.dwLowDateTime;
+        large.u_mut().HighPart = kernel_time.dwHighDateTime;
+        let ktime = *large.QuadPart();
+
+        large.u_mut().LowPart = user_time.dwLowDateTime;
+        large.u_mut().HighPart = kernel_time.dwHighDateTime;
+        let utime = *large.QuadPart();
```

Comment:
it seems sharing some code with `get_thread_time`. how about adding a utility function for this case?